apps/test/unit/lib/levelbuilder/lesson-editor/activitiesEditorReduxTest.js

```diff
@@ -0,0 +1,705 @@
+          isUnplugged: false,
+          isCurrentLevel: false,
+          isConceptLevel: false,
+          named: false,
+          assessment: false,
+          challenge: false,
+          sublevels: [],
+          skin: null,
+          videoKey: null,
+          concepts: '',
+          conceptDifficulty: ''
+        })
+      ).activities;
+      assert.deepEqual(
+        nextState[0].activitySections[2].levels.map(s => s.name),
+        ['Level 1', 'Level 2', 'Level 3']
+      );
+    });
+
+    it('remove level', () => {
+      const nextState = reducer(initialState, removeLevel(1, 3, 1)).activities;
+      assert.deepEqual(
+        nextState[0].activitySections[2].levels.map(s => s.name),
+        ['Level 2']
+      );
+    });
+
+    it('set active variant', () => {
+      const nextState = reducer(initialState, setActiveVariant(1, 3, 1, 2))
+        .activities;
+      assert.equal(nextState[0].activitySections[2].levels[0].activeId, 2);
+    });
+
+    it('set level field', () => {
+      let nextState = reducer(
+        initialState,
+        setField(1, 3, 1, {videoKey: '_a_'})
+      );
+      assert.equal(
+        nextState.activities[0].activitySections[2].levels[0].videoKey,
+        '_a_'
+      );
+      nextState = reducer(nextState, setField(1, 3, 1, {skin: '_b_'}));
+      assert.equal(
+        nextState.activities[0].activitySections[2].levels[0].skin,
+        '_b_'
+      );
+      nextState = reducer(
+        nextState,
+        setField(1, 3, 1, {conceptDifficulty: '_c_'})
+      );
+      assert.equal(
+        nextState.activities[0].activitySections[2].levels[0].conceptDifficulty,
+        '_c_'
+      );
+      nextState = reducer(nextState, setField(1, 3, 1, {concepts: '_d_'}));
+      assert.equal(
+        nextState.activities[0].activitySections[2].levels[0].concepts,
+        '_d_'
+      );
+    });
+  });
+
+  describe('activities', () => {
+    let initialActivities = [];
+
+    beforeEach(() => {
+      initialActivities = [
+        {
+          key: 'x',
+          displayName: 'X',
+          position: 1,
+          activitySections: [
+            {key: 'a', position: 1, displayName: 'A'},
+            {key: 'b', position: 2, displayName: 'B'}
+          ]
+        },
+        {
+          key: 'y',
+          displayName: 'Y',
+          position: 2,
+          activitySections: [
+            {key: 'c', position: 1, displayName: 'C'},
+            {key: 'd', position: 2, displayName: 'D'}
+          ]
+        }
+      ];
+      initialState.activities = initialActivities;
+    });
+
+    it('add activity', () => {
+      const nextState = reducer(initialState, addActivity(3, 'key')).activities;
+      assert.equal(nextState[nextState.length - 1].displayName, '');
+      assert.equal(nextState[nextState.length - 1].key, 'key');
+    });
+
+    it('update activity field', () => {
+      let state = reducer(initialState, updateActivityField(1, 'time', 100));
+      assert.deepEqual(
+        [
```

Comment:
If I'm understanding how these tests are set up correctly, I'm a bit concerned about having so many tests which assert on a literal value for the entire `activities` object:

- If we need to make changes in the future, we'd have to update every single test, right? Even for tests which aren't related to the change.
- When reading the tests, it's hard to know what the specific changes we're expecting from the action are, without memorizing or constantly checking `initialActivities`.